app/modules/perf.c

```diff
@@ -39,7 +38,11 @@ static void ICACHE_RAM_ATTR hw_timer_cb(os_param_t p)
   uint32_t stackaddr;
 
   if (data) {
-    uint32_t pc = *(&stackaddr + data->pc_offset);
+    uint32_t pc;
+    asm (
+      "rsr   %0, EPC1;"    /* read out the EPC */
```

Comment:
Ah yes, of course, interrupts are delivered via hw exceptions too. Brain is in holiday mode, clearly >.>